client/client.go

```diff
@@ -550,18 +642,34 @@ type (
 		// Allows you to update the worker-build-id based version sets for a particular task queue. This is used in
 		// conjunction with workers who specify their build id and thus opt into the feature.
 		// WARNING: Worker versioning is currently experimental
+		// Deprecated: Use [UpdateWorkerVersioningRules] with the versioning-2 api.
 		UpdateWorkerBuildIdCompatibility(ctx context.Context, options *UpdateWorkerBuildIdCompatibilityOptions) error
 
 		// GetWorkerBuildIdCompatibility
 		// Returns the worker-build-id based version sets for a particular task queue.
 		// WARNING: Worker versioning is currently experimental
+		// Deprecated: Use [GetWorkerVersioningRules] with the versioning-2 api.
 		GetWorkerBuildIdCompatibility(ctx context.Context, options *GetWorkerBuildIdCompatibilityOptions) (*WorkerBuildIDVersionSets, error)
 
 		// GetWorkerTaskReachability
 		// Returns which versions are is still in use by open or closed workflows
 		// WARNING: Worker versioning is currently experimental
+		// Deprecated: Use [DescribeTaskQueue] with the versioning-2 api.
 		GetWorkerTaskReachability(ctx context.Context, options *GetWorkerTaskReachabilityOptions) (*WorkerTaskReachability, error)
 
+		// UpdateWorkerVersioningRules
+		// Allows updating the worker-build-id based assignment and redirect rules for a given task queue. This is used in
+		// conjunction with workers who specify their build id and thus opt into the feature.
+		// The errors it can return:
+		//  - serviceerror.FailedPrecondition when the conflict token is invalid
+		// WARNING: Worker versioning-2 is currently experimental, and requires server 1.XX+
+		UpdateWorkerVersioningRules(ctx context.Context, options *UpdateWorkerVersioningRulesOptions) (VersioningConflictToken, error)
```

Comment:
It's very unfortunate that the update call doesn't return the same thing as the get call. I think we should consider changing the API so it does. But if we can't, this should change to returning `UpdateWorkerVersioningRulesResponse` that may contain the conflict token instead of assuming the conflict token is the only thing we ever will return here.

Reply:
Yes, returning the same as get call will save a few roundtrips, let's wait for the server change